lib/runtime/src/pipeline/network/ingress/push_endpoint.rs

```diff
@@ -36,12 +41,22 @@ impl PushEndpoint {
         PushEndpointBuilder::default()
     }
 
-    pub async fn start(self, endpoint: Endpoint) -> Result<()> {
+    pub async fn start(
+        self,
+        endpoint: Endpoint,
+        endpoint_name: String,
```

Comment:
`Endpoint` should provide the `endpoint_name` without it needing to be passed in.  Or are we using a different name?

Reply:
it's odd I agree, but there is no way to get the data out: [Endpoint in async_nats::service::endpoint - Rust](https://docs.rs/async-nats/latest/async_nats/service/endpoint/struct.Endpoint.html)

docs.rsdocs.rs
[Endpoint in async_nats::service::endpoint - Rust](https://docs.rs/async-nats/latest/async_nats/service/endpoint/struct.Endpoint.html)
API documentation for the Rust Endpoint struct in crate async_nats.

Reply:
Apologies. Github's file view is not as smart as rust-analyzer.  When I looked again, `Endpoint` is as you linked.

In that case, we need a representation as passed.  

good enough for v1.

what we'll need to do is pass a scoped object so the task doesn't have access to the full SystemHealth object but scoped down to what it can do and the behaviors it can impact on that object.